cfme/tests/ansible/test_embedded_ansible_basic.py

```diff
@@ -214,6 +215,49 @@ def test_embedded_ansible_repository_invalid_url_crud(request, appliance, wait_f
     repository.delete_if_exists()
 
 
+@pytest.mark.rhel_testing
+@pytest.mark.tier(1)
+def test_embedded_ansible_add_private_repository_crud(
+    request, credentials_collection, appliance, wait_for_ansible
+):
+    """
+    Add Private Repository by using SCM credentials.
+
+    Polarion:
+        assignee: sbulage
+        casecomponent: Ansible
+        caseimportance: critical
+        initialEstimate: 1/6h
+        tags: ansible_embed
+    """
+    creds = {
+        "username": conf.credentials["scm_creds"]["username"],
+        "password": conf.credentials["scm_creds"]["username"],
+    }
+
+    scm_credential = credentials_collection.create(
+        "Scm_credential",
+        "Scm",
+        **creds
+    )
+    repositories = appliance.collections.ansible_repositories
+    repository = repositories.create(
+        name=fauxfactory.gen_alpha(),
+        url="https://gitlab.com/cfme/ansible_private_repo.git",
```

Comment:
Instead of hardcoding can we get it from yaml ?

Reply:
The URL will only used once so there is no special reason to add it into Yamls.

Reply:
I disagree, I think this needs to be pulled from yamls, maybe defaulting to this repo if its not found, instead of skipping the test.

I don't have access to this repo, so if I wanted to run the test, and I had a private repo to test on, I would have to modify the test itself, instead of just the yamls (where I put my private repo credentials), to successfully run this test.

It should be defined and fetched with the scm credentials from the yamls - and the credentials fetching above needs to be wrapped, along with the repo URL, in try/except blocks to call pytest.skip instead of failing with a KeyError/AttributeError during test call phase.

Ideal time to skip would be in setup or collection.  Skipping in collection would require defining pytest_generate_tests for this module, so skipping in setup is reasonable.